### PR TITLE
Feature/user

### PR DESCRIPTION
**Pull request by sas2job**

### CODE REVIEW

File: test/models/user_test.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class UserTest < ActiveSupport::TestCase
+  # test "the truth" do
+  #   assert true
+  # end
+end
```
Comment (alxekb) [inline test/models/user_test.rb]: @sas2job Please remove minitest 🙂

File: test/fixtures/users.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+one:
+  name: MyString
+  email: MyString
+  password: MyString
+
+two:
+  name: MyString
+  email: MyString
+  password: MyString
```
Comment (alxekb) [inline test/fixtures/users.yml]: Please remove fixtires, use factories instead @sas2job

File: test/controllers/api/v1/users_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class Api::V1::UsersControllerTest < ActionDispatch::IntegrationTest
+  # test "the truth" do
+  #   assert true
+  # end
+end
```
Comment (alxekb) [inline test/controllers/api/v1/users_controller_test.rb]: @sas2job same thing here 🙂

**Review by alxekb — CHANGES_REQUESTED** [review]

File: test/models/user_test.rb (hunk 1 above)
Comment (sas2job) [reply test/models/user_test.rb]: It's removed.

File: app/controllers/api/v1/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+          @user.update(user_params)
+          render json: { message: 'User successfully updated.' }
+        else
+          render json: { error: 'Unable to update User.' }
+        end
+      end
+
+      def destroy
+        @user = User.find_by(id: params[:id])
+        if @user
```
Comment (alxekb) [inline app/controllers/api/v1/users_controller.rb]:
```
ApiExceptionHandler.record_not_found if user.blank?

if user.destroy
  render
```

File: app/controllers/api/v1/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        elsif @user.blank?
+          render json: { errors: 'User not found.' }, status: 404
+        else
+          render json: { error: 'Unable to delete User.' }, status: 404
+        end
+      end
+
+      def login
+        @user = User.find_by(email: params[:email])
+        return render json: { errors: 'User not found.' } if @user.class == NilClass
```
Comment (alxekb) [inline app/controllers/api/v1/users_controller.rb]:
```
@user.nil?
```

File: app/controllers/api/v1/users_controller.rb [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        return render json: { errors: 'Password is empty.' } if params[:password] == ''
```
Comment (alxekb) [inline app/controllers/api/v1/users_controller.rb]:
```
unless params[:password].present?
```

**Review by alxekb — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/users_controller.rb (hunk 6 above)
Comment (sas2job) [reply app/controllers/api/v1/users_controller.rb]: Fixed

File: app/controllers/api/v1/users_controller.rb (hunk 5 above)
Comment (sas2job) [reply app/controllers/api/v1/users_controller.rb]: Fixed

**Review by alxekb — APPROVED** [review]
Please add json serializer

**Pull request closed without merging**